helpers/events_filter/events_filter.py

```diff
@@ -42,6 +45,49 @@ def filter_events_by_date_range(query, start_date, end_date):
     return events
 
 
+def validate_calendar_id_input(calendar_id):
+    """
+    Ensures that the calendar id is supplied
+    """
+    if not calendar_id:
+        raise GraphQLError("Calendar Id missing")
+
+
+def filter_event_by_room(query, calendar_id, start_date, end_date):
+    """
+    Returns all events in a room using the calendar id as the filter field.
+    If the start and end dates are provided, it considers them as well.
+    """
+    validate_calendar_id_input(calendar_id)
+    room = EventsModel.query.join(RoomModel).filter(
+                RoomModel.calendar_id == calendar_id
+            ).first()
+    if not room:
+            raise GraphQLError("No events with the given CalendarId")
+    room_id = room.room_id
+    if not start_date and not end_date:
+        events = query.filter(
+                EventsModel.room_id == room_id,
+                EventsModel.state == 'active'
+            ).all()
+        if not events:
+            raise GraphQLError('Events do not exist'+ end_date)
+        return events
+
```

Comment:
I have failed to come up with a scenario where start_date and end_date are not provided and then have Events do not exist as an error from GraphQLError. because without them you can still have the results if the calendar_Id is valid based on your logic here.